tardis/energy_input/tests/test_gamma_ray_transport.py

```diff
@@ -0,0 +1,195 @@
+    desired = (volume * density).to(u.g).value
+
+    shell_masses = calculate_shell_masses(model).value
+    npt.assert_almost_equal(shell_masses, desired)
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56"])
+def test_activity(simulation_setup, nuclide_name):
+    """
+    Function to test the decay of 56Ni in radioactivedecay with an analytical solution.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+    nuclide = rd.Nuclide(nuclide_name)
+    model = simulation_setup
+    t_half = nuclide.half_life() * u.s
+    decay_constant = np.log(2) / t_half
+    time_delta = 1.0 * u.s
+    shell_masses = calculate_shell_masses(model)
+    raw_isotope_abundance = model.raw_isotope_abundance
+    raw_isotope_abundance_mass = raw_isotope_abundance.apply(
+        lambda x: x * shell_masses, axis=1
+    )
+    mass = raw_isotope_abundance_mass.loc[nuclide.Z, nuclide.A][0]
+    iso_dict = create_isotope_dicts(raw_isotope_abundance, shell_masses)
+    inv_dict = create_inventories_dict(iso_dict)
+
+    total_decays = calculate_total_decays(inv_dict, time_delta)
+    actual = total_decays[0][nuclide.Z, nuclide.A][nuclide_name]
+
+    isotopic_mass = nuclide.atomic_mass * (u.g)
+    number_of_moles = mass * (u.g) / isotopic_mass
+    number_of_atoms = (number_of_moles * c.N_A).value
+    N1 = number_of_atoms * np.exp(-decay_constant * time_delta)
+    expected = number_of_atoms - N1.value
+
+    npt.assert_allclose(actual, expected, rtol=1e-7)
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56"])
+def test_activity_chain(simulation_setup, nuclide_name):
+    """
+    Function to test the decay of a two atom decay chain in radioactivedecay with an analytical solution.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+    nuclide = rd.Nuclide(nuclide_name)
+    model = simulation_setup
+    t_half = nuclide.half_life()
+    decay_constant = np.log(2) / t_half
+    time_delta = 1.0 * (u.d).to(u.s)
+    progeny = rd.Nuclide(nuclide.progeny()[0])
+    decay_constant_progeny = np.log(2) / progeny.half_life()
+    shell_masses = calculate_shell_masses(model)
+    raw_isotope_abundance = model.raw_isotope_abundance
+    raw_isotope_abundance_mass = raw_isotope_abundance.apply(
+        lambda x: x * shell_masses, axis=1
+    )
+    mass = raw_isotope_abundance_mass.loc[nuclide.Z, nuclide.A][0]
+    iso_dict = create_isotope_dicts(raw_isotope_abundance, shell_masses)
+    inv_dict = create_inventories_dict(iso_dict)
+    total_decays = calculate_total_decays(inv_dict, time_delta)
+    actual_parent = total_decays[0][nuclide.Z, nuclide.A][nuclide_name]
+    actual_progeny = total_decays[0][nuclide.Z, nuclide.A][nuclide.progeny()[0]]
+    isotopic_mass = nuclide.atomic_mass
+    number_of_moles = mass / isotopic_mass
+    number_of_atoms = number_of_moles * 6.02214076e23
```

Comment:
Use the astropy constant here instead of hardcoding the number.

Reply:
Astropy constant has not being used in radioactive decay.